packages/core/src/app/shipping/stripeUPE/StripeShippingAddress.tsx

```diff
@@ -101,20 +101,21 @@ const StripeShippingAddress: FunctionComponent<StripeShippingAddressProps> = (pr
 
     const handleStripeShippingAddress = useCallback(async (shipping: StripeShippingEvent) => {
         const {complete, value: { address = { country: '', state: '', line1: '', line2: '', city: '', postal_code: '' }
-            , name = '' } } = shipping;
+            , name = '', firstName = '', lastName = '' } } = shipping;
 
         if(complete) {
             if (step.isComplete || (shippingAddress?.firstName && shipping.isNewAddress)) {
                 handleLoading();
             }
 
-            const names = name.split(' ');
+            const names = name?.split(' ');
+
             // @ts-ignore
             const country = availableShippingList?.find(country => country.code === address.country).name;
             const state = StripeStateMapper(address.country, address.state);
             const shippingValue = {
-                firstName: names[0],
-                lastName: names[1] || ' ',
+                firstName: firstName || names[0],
+                lastName: lastName || names[1],
```

Comment:
Nice work, will we be removing the names[0]/[1] approach in future?

Reply:
Thank you, it would be better to have them as a fall back, if you have display split or full the variable name will always come with data so it is a useful just in case the split name fails.